torch/distributed/_tensor/_dispatch.py

```diff
@@ -224,7 +224,12 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             if output_sharding.output_spec is not None:
                 return args[0]
             else:
```

Comment:
If we really want zero changes in grad scaler, it's possible to do all-reduce(found_inf) in DTensor dispatch. For `aten.equal.default`, we did the same thing

if we really want to avoid all-reduce(found_inf) in DTensor dispatch, we can modify grad_scaler.py in two places
* found_inf should be initialized with `grad.new_full()` instead of `torch.Tensor(0)`, so found_inf becomes `DTensor` when `.grad` is DTensor
* before optimizer step, need a way to redistribute `found_inf` from `Partial(max)` to `Replicate` without if-else base on DTensor in  grad_scaler.py